src/main.rs

```diff
@@ -174,20 +174,38 @@ impl State {
 async fn main() -> anyhow::Result<()> {
     tracing_subscriber::fmt::init();
 
-    let mut processed_args = Vec::new();
-    for arg in std::env::args() {
-        match arg.strip_prefix('@') {
-            None => processed_args.push(arg),
-            Some(path) => {
-                let config = read_to_string(path).context("Failed to read config file")?;
-                for line in config.lines() {
-                    processed_args.push(line.to_string());
+    let args = std::env::args()
+        .try_fold(Vec::new(), |mut args, arg| {
+            if let Some(path) = arg.strip_prefix('@') {
+                let conf = read(path).context(format!("failed to read config file at `{path}`"))?;
+                match toml::from_slice(&conf)
+                    .context(format!("failed to parse config file at `{path}` as TOML"))?
+                {
+                    toml::Value::Table(kv) => kv.into_iter().try_for_each(|(k, v)| {
+                        match v {
+                            toml::Value::String(v) => args.push(format!("--{k}={v}")),
+                            toml::Value::Integer(v) => args.push(format!("--{k}={v}")),
+                            toml::Value::Float(v) => args.push(format!("--{k}={v}")),
+                            toml::Value::Boolean(v) => {
+                                if v {
+                                    args.push(format!("--{k}"))
+                                }
+                            }
+                            _ => bail!(
+                                "unsupported value type for field `{k}` in config file at `{path}`"
+                            ),
+                        }
+                        Ok(())
+                    })?,
+                    _ => bail!("invalid config file format in file at `{path}`"),
                 }
+            } else {
+                args.push(arg);
             }
-        }
-    }
-
-    let args = Args::parse_from(processed_args);
+            Ok(args)
+        })
+        .map(Args::parse_from)
```

Comment:
Is this still using `clap`? Otherwise, can it be removed from the struct and `Cargo.toml`?

Reply:
Yes, it's a hacky workaround read a TOML into an "argv" and pass it to clap. We should definitely have something more robust long-term, but this should be fine for MVP. See also discussion in https://github.com/profianinc/drawbridge/pull/215 for more details